From: Jan Engelhardt <jengelh@inai.de>
Date: 2022-04-02 13:01:22.046234056 +0200

Add missing functions in the 'else' part of a if..endif block
differentiating between C++ version.

opal makes use of this by way of ptlib's threadpool.h's PSafeWorkArg2.

---
 include/ptlib/object.h |    2 ++
 1 file changed, 2 insertions(+)

Index: ptlib-2.18.8/include/ptlib/object.h
===================================================================
--- ptlib-2.18.8.orig/include/ptlib/object.h
+++ ptlib-2.18.8/include/ptlib/object.h
@@ -109,6 +109,8 @@ using namespace std; // Not a good pract
     public:
       PAutoPtr() = default;
       explicit PAutoPtr(T * p) : std::unique_ptr<T>(p) { }
+      PAutoPtr(PAutoPtr &other) : std::unique_ptr<T>(std::move(other)) {}
+      PAutoPtr(PAutoPtr &&other) : std::unique_ptr<T>(std::move(other)) {}
       void transfer(PAutoPtr & other) { std::unique_ptr<T>::operator=(std::move(other)); }
   };
 #endif
